sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/http/rest/RestProxyUtils.java

```diff
@@ -95,12 +96,73 @@ public static Context mergeRequestOptionsContext(Context context, RequestOptions
         Context optionsContext = options.getContext();
 
         if (optionsContext != null && optionsContext != Context.NONE) {
-            context = Context.mergeContexts(context, optionsContext);
+            context = mergeContexts(context, optionsContext);
         }
 
         return context;
     }
 
+    /**
+     * Merges data from two {@link Context} instances.
+     *
+     * @param into The {@link Context} to merge data into.
+     * @param from The {@link Context} to merge data from.
+     *
+     * @return The merged {@link Context}.
+     */
+    private static Context mergeContexts(Context into, Context from) {
```

Comment:
Do we need this method if it's internal? We can just inline this in the above method.

Reply:
Not really, I kept it separate to make it easier to read and in case we ever wanted to grab this and put it out as public API again.